helperApp/sift-science/sift.controller.js

```diff
@@ -0,0 +1,36 @@
+
+function SiftController ($scope, $routeParams, $timeout) {
+  const isInIframe = (window.parent !== window);
+
+  if (!isInIframe) {
+    console.error('Must be inside iframe');
+    return;
+  }
+
+  const parentUrl = document.referrer.replace('/wallet/', '');
+
+  var _sift = window._sift = window._sift || [];
+  _sift.push(['_setCookieDomain', window.location.hostname]);
+  _sift.push(['_setAccount', $routeParams.apiKey]);
+  _sift.push(['_setUserId', $routeParams.userId]);
+  _sift.push(['_trackPageview']);
+
+  // Sift Science submits data by loading a GIF image. We need to detect when
+  // it's done loading.
+  var OriginalImage = window.Image;
+  window.Image = function (width, height) {
+    let img = new OriginalImage(width, height);
+
+    // onload is overriden by Sift Science directly after new Image(1,1)
+    $timeout(() => {
+      let originalOnLoad = img.onload;
+      img.onload = () => {
+        window.parent.postMessage({from: 'sift-science', to: 'exchange', command: 'done'}, parentUrl);
+        originalOnLoad();
+      };
+    }, 0);
+    return img;
+  };
```

Comment:
A little unconventional, but works and I don't know a better way to do this. I guess the alternative would be to leave it as long as the user is in the buy screen.